src/java/src/main/java/ai/onnxruntime/genai/Model.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT License.
+ */
+package ai.onnxruntime.genai;
+
+public final class Model implements AutoCloseable {
+  private long nativeHandle;
+
+  public Model(String modelPath) throws GenAIException {
+    nativeHandle = createModel(modelPath);
+  }
+
+  /**
+   * Creates a Tokenizer instance for this model. The model contains the configuration information
+   * that determines the tokenizer to use.
+   *
+   * @return The Tokenizer instance.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
+  public Tokenizer createTokenizer() throws GenAIException {
+    if (nativeHandle == 0) {
+      throw new IllegalStateException("Instance has been freed and is invalid");
+    }
+
+    return new Tokenizer(this);
+  }
+
+  // NOTE: Having model.createGeneratorParams is still under discussion.
+  // model.createTokenizer is consistent with the python setup at least and agreed upon.
+
+  /**
+   * Creates a GeneratorParams instance for executing the model. NOTE: GeneratorParams internally
+   * uses the Model, so the Model instance must remain valid
+   *
+   * @return The GeneratorParams instance.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
+  public GeneratorParams createGeneratorParams() throws GenAIException {
```

Comment:
I have both the tokenizer and the generator params creation being done via the Model class currently. The rationale is that creating those classes involves Model internals (the config), hence we pass the model to the ctor of each. It's also more consistent this way vs creating the tokenizer via the model but the generator params directly. 

Happy to discuss further and adjust the other language bindings to be consistent.

Reply:
I think this is ok for now. We need to standardize the API. This can happen as a separate PR.